subprojects/core/src/test/groovy/org/gradle/api/internal/AbstractDomainObjectCollectionSpec.groovy

```diff
@@ -47,6 +48,12 @@ abstract class AbstractDomainObjectCollectionSpec<T> extends Specification {
         return elements
     }
 
+    abstract boolean isExternalProviderAllowed()
```

Comment:
I don't think we need to change this, but this might have been better handled through inheritance - e.g. make the other test classes inherit from an `ExternalProviderAllowedDomainObjectCollectionTest` and move the relevant methods down into that class, meanwhile `DefaultTaskContainerTest`would just extend directly off of this class.

I would only resort to this sort of thing if the existing class structure would not allow another solution.